### PR TITLE
Dependency fix

### PR DESCRIPTION
**Pull request by jgonyea**
An initial stab at ensuring that already installed (but disabled) dependencies are enabled.

If you feel like I should split out the enabling to its own command file, I can refactor and resubmit.

This PR also contains some PSR2 fixes.

### CODE REVIEW

**Comment by jgonyea** [comment]
Forgot to mention that this is related to Issue #2383 https://github.com/getgrav/grav/issues/2383

**Comment by OleVik** [comment]
Just to clarify; the change enforces an enabled-state for all plugins when upgrading/updating?

If so, I wholeheartedly agree that it is a necessary constraint. Otherwise, upgrading/updating can potentially push versions beyond the point where a disabled plugin can operate. The disabled-state may only be temporary, and so upgrading can break functionality without the possibility of a downgrade to the previous ecosystem of working versions.

File: system/src/Grav/Console/Gpm/InstallCommand.php [hunk 1]
```diff
@@ -322,6 +345,42 @@ public function installDependencies($dependencies, $type, $message, $required =
...
+     *
+     * @param array  $package_slug   The package to enable.
+     *
+     * @return bool Success of enabling a package.
+     * 
+     * @throws \Exception
+     */
+    public static function enablePackage($package_slug)
+    {
+        $grav = new Grav;
```
Comment (w00fz) [inline system/src/Grav/Console/Gpm/InstallCommand.php]:
```suggestion
        $grav = new Grav;
```

File: system/src/Grav/Console/Gpm/InstallCommand.php [hunk 2]
```diff
@@ -322,6 +345,42 @@ public function installDependencies($dependencies, $type, $message, $required =
...
+
+        $config_file_path = USER_DIR . "config" . DS . "plugins" . DS;
+        $config_filename = $package_slug . '.yaml';
+        $config_file = File::instance($config_file_path . $config_filename);
+        
+        if (!is_writable($config_file->filename())) {
+            $message = "Could not save config for " . $package_slug . ". Is the config folder/ file writable?";
+            $grav::instance()['log']->error($message);
```
Comment (w00fz) [inline system/src/Grav/Console/Gpm/InstallCommand.php]:
```suggestion
            $grav['log]->error($message);
```

File: system/src/Grav/Console/Gpm/InstallCommand.php [hunk 3]
```diff
@@ -322,6 +345,42 @@ public function installDependencies($dependencies, $type, $message, $required =
...
+        if ($config_file->exists()) {
+            // Load existing config file.
+            $config_data = Yaml::parse($config_file->content());
+        }
+
+        // Set 'enable' plugin option.
+        $config_data['enabled'] = true;
+
+        $config_file->save(Yaml::dump($config_data));
+        $grav::instance()['log']->info($package_slug . " was enabled.");
```
Comment (w00fz) [inline system/src/Grav/Console/Gpm/InstallCommand.php]:
```suggestion
        $grav['log']->info($package_slug . " was enabled.");
```

**Comment by jgonyea** [comment]
This only forces an enable of Plugin A if: 

1. Plugin A is installed, but in a disabled state
2. Another plugin/ theme (Package B) is installed/ updated via GPM that depends on that disabled Plugin A


I didn't go as far as forcing a enabled yaml file for even the installed Package B.

File: system/src/Grav/Console/Gpm/InstallCommand.php [hunk 4]
```diff
@@ -322,6 +345,42 @@ public function installDependencies($dependencies, $type, $message, $required =
...
+            return false;
+        }
+
+        if ($config_file->exists()) {
+            // Load existing config file.
+            $config_data = Yaml::parse($config_file->content());
+        }
+
+        // Set 'enable' plugin option.
+        $config_data['enabled'] = true;
```
Comment (w00fz) [inline system/src/Grav/Console/Gpm/InstallCommand.php]: This could fail if `$config_file` doesn't exist?

Comment (jgonyea) [reply system/src/Grav/Console/Gpm/InstallCommand.php]: No, if I change L366 to:
`if (!$config_file->writable($config_file->filename())) {`

Comment (jgonyea) [reply system/src/Grav/Console/Gpm/InstallCommand.php]: To step through the steps just prior:
if the config file is present, then the `Yaml::parse` loads its contents into the $config_data array.
Whether or not the config file isn't present, then this L377 will just create a new array (or edit the existing array) at $config_data with the enable key setting it to `true`

**Comment by jgonyea** [comment]
Newest commit now includes removal of a package's config yaml on uninstall.

**Comment by jgonyea** [comment]
I'm closing this pull request and will resubmit a better, updated version of it.

**Pull request closed without merging**